File: 2.Lista/NuevaLista.py
class Nodo:
    def __init__(self, dato):
        self.dato = dato
        self.siguiente = None

class Lista:
    def __init__(self):
        self.inicio = None

    def agregar(self, dato):
        if not self.inicio:
            self.inicio = Nodo(dato)
        else:
            aux = self.inicio
            while aux.siguiente:
                aux = aux.siguiente
            aux.siguiente = Nodo(dato)

    def listar(self):
        aux = self.inicio
        while aux:
            print(aux.dato)
            aux = aux.siguiente

File: 2.Lista/NuevaLista.java
class Nodo {
    int dato;
    Nodo siguiente;

    public Nodo(int dato) {
        this.dato = dato;
        this.siguiente = null;
    }
}

class Lista {
    Nodo inicio;

    public Lista() {
        this.inicio = null;
    }

    public void agregar(int dato) {
        if (inicio == null) {
            inicio = new Nodo(dato);
        } else {
            Nodo aux = inicio;
            while (aux.siguiente != null) {
                aux = aux.siguiente;
            }
            aux.siguiente = new Nodo(dato);
        }
    }

    public void listar() {
        Nodo aux = inicio;
        while (aux != null) {
            System.out.println(aux.dato);
            aux = aux.siguiente;
        }
    }
}


